Source/WebCore/platform/graphics/mac/controls/SwitchMacUtilities.mm

```diff
@@ -77,6 +77,14 @@ static FloatRect rectForBounds(const FloatRect& bounds)
     return (__bridge NSString *)kCUISizeRegular;
 }
 
+static float easeInOut(const float progress)
+{
+    // FIXME: this should be a MESSAGE_CHECK, ideally added to
+    // WebCoreArgumentCoders.serialization.in when it gains that capability.
+    auto clampedProgress = clampTo<float>(progress, 0.0f, 1.0f);
```

Comment:
`easeInOut` is too far removed from the IPC layer to be worrying about validation.

Can't you add a `[Validator=...]` in `WebCoreArgumentCoders.serialization.in`?

If not, I would still move the clamping earlier, like in the constructor of the part.